### PR TITLE
Upgrade to v1.138.0 of the AliCloud Terraform Provider

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #200
Fixes: #201
Fixes: #202
Fixes: #203﻿
Fixes: #205
Fixes: #206

### CODE REVIEW

File: provider/resources.go [hunk 1]
```diff
@@ -48,15 +48,19 @@ const (
...
 	cloudFirewallMod       = "CloudFirewall"
 	cloudStorageGatewayMod = "CloudStorageGateway"
+	cloudSsoMod            = "CloudSso"
 	cmsMod                 = "Cms"
 	cfgMod                 = "Cfg"
 	crMod                  = "CR"
 	csMod                  = "CS"
 	datahubMod             = "Datahub"
 	dataWorksMod           = "DataWorks"
+	databseFilesystemMod   = "DatabaseFilesystem"
```
Comment (mikhailshilkov) [inline provider/resources.go:62]:
```suggestion
	databaseFilesystemMod   = "DatabaseFilesystem"
```

File: provider/resources.go [hunk 2]
```diff
@@ -335,6 +368,7 @@ func Provider() tfbridge.ProviderInfo {
 			"alicloud_cms_group_metric_rule":       {Tok: resource(cmsMod, "GroupMetricRule")},
 			"alicloud_cms_monitor_group":           {Tok: resource(cmsMod, "MonitorGroup")},
 			"alicloud_cms_monitor_group_instances": {Tok: resource(cmsMod, "MonitorGroupInstances")},
+			"alicloud_cms_metric_rule_template":    {Tok: resource(cmsMod, "MetricRuleTemlate")},
```
Comment (mikhailshilkov) [inline provider/resources.go:371]:
```suggestion
			"alicloud_cms_metric_rule_template":    {Tok: resource(cmsMod, "MetricRuleTemplate")},
```

File: provider/resources.go [hunk 3]
```diff
@@ -982,13 +1099,17 @@ func Provider() tfbridge.ProviderInfo {
...
+			"alicloud_cloud_sso_directories": {Tok: dataSource(cloudSsoMod, "getDirectories")},
+
 			// CMS
 			"alicloud_cms_alarm_contacts":            {Tok: dataSource(cmsMod, "getAlarmContacts")},
 			"alicloud_cms_alarm_contact_groups":      {Tok: dataSource(cmsMod, "getAlarmContactGroups")},
 			"alicloud_cms_group_metric_rules":        {Tok: dataSource(cmsMod, "getGroupMetricRules")},
 			"alicloud_cms_service":                   {Tok: dataSource(cmsMod, "getService")},
 			"alicloud_cms_monitor_groups":            {Tok: dataSource(cmsMod, "getMonitorGroups")},
 			"alicloud_cms_monitor_group_instanceses": {Tok: dataSource(cmsMod, "getMonitorGroupInstances")},
+			"alicloud_cms_metric_rule_templates":     {Tok: dataSource(cmsMod, "getRuleTemplates")},
```
Comment (mikhailshilkov) [inline provider/resources.go:1112]:
```suggestion
			"alicloud_cms_metric_rule_templates":     {Tok: dataSource(cmsMod, "getMetricRuleTemplates")},
```

**Review by mikhailshilkov — COMMENTED** [review]
A few suggestions

File: provider/resources.go [hunk 4]
```diff
@@ -862,18 +990,27 @@ func Provider() tfbridge.ProviderInfo {
...
-			"alicloud_regions":             {Tok: dataSource(alicloudMod, "getRegions")},
-			"alicloud_caller_identity":     {Tok: dataSource(alicloudMod, "getCallerIdentity")},
-			"alicloud_file_crc64_checksum": {Tok: dataSource(alicloudMod, "getFileCrc64Checksum")},
-			"alicloud_msc_sub_contacts":    {Tok: dataSource(alicloudMod, "getMscSubContracts")},
+			"alicloud_account":               {Tok: dataSource(alicloudMod, "getAccount")},
+			"alicloud_zones":                 {Tok: dataSource(alicloudMod, "getZones")},
+			"alicloud_regions":               {Tok: dataSource(alicloudMod, "getRegions")},
+			"alicloud_caller_identity":       {Tok: dataSource(alicloudMod, "getCallerIdentity")},
+			"alicloud_file_crc64_checksum":   {Tok: dataSource(alicloudMod, "getFileCrc64Checksum")},
+			"alicloud_msc_sub_contacts":      {Tok: dataSource(alicloudMod, "getMscSubContracts")},
```
Comment (lblackstone) [inline provider/resources.go:1012]:
```suggestion
			"alicloud_msc_sub_contacts":      {Tok: dataSource(alicloudMod, "getMscSubContacts")},
```

File: provider/resources.go [hunk 5]
```diff
@@ -1392,9 +1602,16 @@ func Provider() tfbridge.ProviderInfo {
...
+			"alicloud_vpc_nat_ip_cidrs":             {Tok: dataSource(vpcMod, "getNatIpCidrs")},
+			"alicloud_vpc_nat_ips":                  {Tok: dataSource(vpcMod, "getNatIps")},
+
+			// Vod
+			"alicloud_vod_domains": {Tok: dataSource(vodMod, "getDomains")},
 
 			// Video Surveillance
-			"alicloud_vs_service": {Tok: dataSource(vsMod, "getService")},
+			"alicloud_vs_service":                       {Tok: dataSource(vsMod, "getService")},
+			"alicloud_video_surveillance_system_groups": {Tok: dataSource(vsMod, "getSystemGroups")},
```
Comment (lblackstone) [inline provider/resources.go:1614]: Should this be `alicloud_vs_system_groups`?

**Review by lblackstone — CHANGES_REQUESTED** [review]

File: provider/resources.go (hunk 5 above)
Comment (stack72) [reply provider/resources.go:1614]: Nah, they broke their naming convention :D So _vs_ is still _video_surveillance_ :/

File: provider/resources.go (hunk 4 above)
Comment (stack72) [reply provider/resources.go:1012]: Great catch! thanks

**Review by lblackstone — APPROVED** [review]

**Pull request merged**